lib/Onyx.js

```diff
@@ -794,6 +802,36 @@ function notifyCollectionSubscribersOnNextTick(key, value) {
     Promise.resolve().then(() => keysChanged(key, value));
 }
 
+/**
+ * This method waits for prior modifications to complete before running the next modification.
+ *
+ * @param {Function} nextModification
+ * @param {String} key If "key" is not set then the modification will be queued after all prior modifications completed.
+ * @returns {Promise}
+ */
+function waitPendingWrites(nextModification, key) {
+    // If key is not set, then we are doing a modification that affects all keys. We must wait for all prior modifications to complete.
```

Comment:
There is a _very specific_ single case where we do this... we are making things harder to understand by alluding to the "modification that affects all keys".

Reply:
Do you mean, that we should extract this logic (when key is not defined) to the `clear` function itself? Or rather just change the documentation?

Reply:
I personally would extract it. Or add something like `isClearingStorage: bool` argument to be explicit. Then the code documents itself.